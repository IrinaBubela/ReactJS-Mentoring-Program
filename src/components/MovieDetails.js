import React from 'react';

const MovieDetails = ({ movie }) => {
  return (
    <div className="container mt-4">
      <div className="row movie-details">
        <div className="col-md-4">
          <div className="poster">
            <img src={movie.imageUrl} alt={movie.name} />
          </div>
        </div>
        <div className="col-md-8">
          <div className="details">
            <h2>{movie.name}</h2>
            <p><strong>Release Year:</strong> {movie.releaseYear}</p>
            <p><strong>Rating:</strong> {movie.rating}</p>
            <p><strong>Duration:</strong> {movie.duration}</p>
            <p><strong>Description:</strong> {movie.description}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MovieDetails;
