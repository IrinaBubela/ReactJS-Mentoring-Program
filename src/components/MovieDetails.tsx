import React from 'react';
import { Movie } from './MovieListPage';

interface Props {
    movie: Movie;
}

const MovieDetails: React.FC<Props> = ({ movie }) => {
  return (
    <div className="container mt-4">
      <div className="row movie-details">
        <div className="col-md-4">
          <div className="poster">
            <img src={movie.poster_path} alt={movie.title} />
          </div>
        </div>
        <div className="col-md-8">
          <div className="details">
            <h2>{movie.title}</h2>
            <p><strong>Release Year:</strong> {movie.release_date}</p>
            <p><strong>Rating:</strong> {movie.vote_average}</p>
            <p><strong>Duration:</strong> {movie.runtime} min</p>
            <p><strong>Description:</strong> {movie.overview}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MovieDetails;
