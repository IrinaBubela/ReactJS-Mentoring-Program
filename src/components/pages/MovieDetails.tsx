import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { Movie } from '.';

const MovieDetails: React.FC = () => {
  const { movieId } = useParams();
  const [movieDetails, setMovieDetails] = useState<Movie>();
  const [error, setError] = useState('');

  useEffect(() => {
    fetch(`http://localhost:4000/movies/${movieId}`)
      .then((response) => response.json())
      .then((data: Movie) => setMovieDetails(data))
      .catch((err) => setError(err));
  }, []);

  return (
    <div className="container">
      {movieDetails ? (
        <div className="row movie-details">
          <div className="col-md-4">
            <div className="poster">
              <img src={movieDetails.poster_path} alt={movieDetails.title} />
            </div>
          </div>
          <div className="col-md-8">
            <div className="details">
              <h2>{movieDetails.title}</h2>
              <p><strong>Release Year:</strong> {movieDetails.release_date}</p>
              <p><strong>Rating:</strong> {movieDetails.vote_average}</p>
              <p><strong>Duration:</strong> {movieDetails.runtime} min</p>
              <p><strong>Description:</strong> {movieDetails.overview}</p>
            </div>
          </div>
        </div>
      ) : (<div>{error}</div>)
      };
    </div >
  );
};

export default MovieDetails;
