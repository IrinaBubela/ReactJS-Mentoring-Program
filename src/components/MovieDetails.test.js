import React from 'react';
import { render } from '@testing-library/react';
import MovieDetails from './MovieDetails';

// Mock movie data
const movie = {
  imageUrl: 'example.jpg',
  name: 'Example Movie',
  releaseYear: 2021,
  rating: 'PG-13',
  duration: '2h',
  description: 'This is an example movie description.'
};

describe('MovieDetails', () => {
  it('renders correctly', () => {
    const { getByText, getByAltText } = render(<MovieDetails movie={movie} />);

    // Check if movie details are rendered correctly
    expect(getByText('Example Movie')).toBeInTheDocument();
    expect(getByText('Release Year:')).toBeInTheDocument();
    expect(getByText('2021')).toBeInTheDocument();
    expect(getByText('Rating:')).toBeInTheDocument();
    expect(getByText('PG-13')).toBeInTheDocument();
    expect(getByText('Duration:')).toBeInTheDocument();
    expect(getByText('2h')).toBeInTheDocument();
    expect(getByText('Description:')).toBeInTheDocument();
    expect(getByText('This is an example movie description.')).toBeInTheDocument();

    // Check if movie poster is rendered correctly
    const moviePoster = getByAltText('Example Movie');
    expect(moviePoster).toBeInTheDocument();
    expect(moviePoster).toHaveAttribute('src', 'example.jpg');
  });
});
