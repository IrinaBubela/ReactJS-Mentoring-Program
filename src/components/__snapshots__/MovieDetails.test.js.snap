// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div />
    ;
  </div>
</div>
`;
